frontend/src/components/pages/QuestionPage.tsx

```diff
@@ -0,0 +1,55 @@
+import {useParams} from "react-router-dom";
+import {useEffect, useState} from "react";
+import axios from "axios";
+import {Question} from "../../types/types.ts";
+import OptionCard from "../main/OptionCard.tsx";
+
+function QuestionPage() {
+    const {topicId} = useParams();
+    const [questions, setQuestions] = useState<Question[]>()
+    const [selectedOption, setSelectedOption] = useState<string | null>(null);
+    const optionCount:string[] = ["A","B","C","D"];
+
+    useEffect(() => {
+        if(topicId) {
+            axios.get<Question[]>("/api/questions/" + topicId)
+                .then(response => setQuestions(response.data))
+                .catch(error => console.log("error", error))
+        }
+    }, [topicId])
+
+    function handleClick(option:string) {
+        setSelectedOption(option);
+    }
+
+    function handleSubmit() {
+        alert(selectedOption);
+        setSelectedOption(null)
+    }
+
+    return questions && (
+        <main className={"main-content"}>
+            <section className={"info-section"}>
+                <p className={"info-text"}>Question 6 of 10</p>
+                <h2>{questions[0].query}</h2>
+                <div className={"progress-bar"}></div>
+            </section>
+
+            <section className={"options-container"}>
+                {
+                    questions[0].options.map((option, index) => {
+                    return <OptionCard
+                        key={questions[0]+option}
+                        optionCount={optionCount[index]}
```

Comment:
This variable needs to be renamed. It is not self explaining what it stores. A better fit would be answerChoice or quizAnswer
```suggestion
                        quizAnswer={optionCount[index]}
```

Reply:
I renamed optionCount to quizAnswer in all related components, types and CSS styles